api/src/model/play_asset_pack_state.gd

```diff
@@ -0,0 +1,65 @@
+# ##############################################################################
+#
+#	Copyright 2020 Google LLC
+#
+#	Licensed under the Apache License, Version 2.0 (the "License");
+#	you may not use this file except in compliance with the License.
+#	You may obtain a copy of the License at
+#
+#		https://www.apache.org/licenses/LICENSE-2.0
+#
+#	Unless required by applicable law or agreed to in writing, software
+#	distributed under the License is distributed on an "AS IS" BASIS,
+#	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#	See the License for the specific language governing permissions and
+#	limitations under the License.
+#
+# ##############################################################################
+#
+# Wraps Play Core's AssetPackState which represents the state of an individual 
+# asset pack.
+#
+# ##############################################################################
+class_name PlayAssetPackState
+extends Object
+
+var _asset_pack_state_dict : Dictionary
+
+func _init(init_dictionary : Dictionary):
+	_asset_pack_state_dict = init_dictionary.duplicate()
+
+# -----------------------------------------------------------------------------
+# Returns the name of the pack.
+# -----------------------------------------------------------------------------
+func get_name() -> String:
+	return _asset_pack_state_dict["name"]
+
+# -----------------------------------------------------------------------------
+# Returns the download status of the pack.
+# -----------------------------------------------------------------------------
+func get_status() -> int:
+	return _asset_pack_state_dict["status"]
+
+# -----------------------------------------------------------------------------
+# Returns the error code for the pack, if Play has failed to download the pack.
+# -----------------------------------------------------------------------------
+func get_error_code() -> int:
+	return _asset_pack_state_dict["errorCode"]
+
+# -----------------------------------------------------------------------------
+# Returns the total number of bytes already downloaded for the pack.
+# -----------------------------------------------------------------------------
+func get_bytes_downloaded() -> int:
```

Comment:
This ```int``` datatype is intentional since GDScript has no builtin long type, and [builtin Integer type](https://docs.godotengine.org/en/stable/classes/class_int.html) for Godot is already 64-bit signed int. I tested passing in Java int/long typed numbers (with arbitrary values, plus ```MAX_VALUE``` and ```MIN_VALUE```) to Godot runtime and all of them can be correctly casted to GDScript int.

Since this is a project wide design decision I think it is best to make a not/comment. What is the best place I can put this? Should I put it in the code base or in my design document?

Reply:
Design doc would be a great place to put it. I'm guessing developers familiar with Godot will know this so it's not something we need to call out to them.

Reply:
Thanks for the suggestion! I added this design decision [mention](https://docs.google.com/document/d/1Y8ps6pqrsICvM_0FneecPl96k6G1PMJzAcDSFfu9Jq4/edit?disco=AAAAJ_XActU) in the design doc.